design/aws-ecs/aws-ecs-multiple-target-groups-support.md

```diff
@@ -0,0 +1,641 @@
+        containerPorts: ports
+      });
+    }
+    return registeredContainers;
+  }
+
+  ...
+
+}
+```
+
+``` ts
+/**
+ * The base class for Ec2Service and FargateService services.
+ */
+export abstract class BaseService extends Resource
+  implements IService, elbv2.IApplicationLoadBalancerTarget, elbv2.INetworkLoadBalancerTarget {
+
+  ...
+
+  /**
+   * Validate the existence of the target in the taskDefinition.
+   */
+  private validateExist(target: ContainerTarget): boolean {
+    for (const container of this.taskDefinition.containerNamePorts) {
+      if (target.containerName === container.containerName) {
+        if (container.containerPorts.includes(target.containerPort)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+}  
+```
+
+### Part 2: Add interfaces for new API
+
+``` ts
+/**
+ * Properties for defining an ECS target.
+ */
+export interface ContainerTarget {
+  /**
+   * The name of the container.
+   */
+  readonly containerName: string;
+
+  /**
+   * The port number of the container.
+   */
+  readonly containerPort: number;
+}
+
+/**
+ * Properties for ECS container targets.
+ */
+export interface TargetProps {
+  /**
+   * Name of the container.
+   */
+  readonly containerName: string,
+  /**
+   * Container targets of the container.
+   */
+  readonly containerTargets: ContainerTargetProps[]
+}
+
+/**
+ * Properties for registering a container target.
+ */
+export interface ContainerTargetProps {
+  /**
+   * Container port number to be registered.
+   */
+  readonly containerPort: number,
+  /**
+   * Setting to create and attach target groups.
+   */
+  readonly targetGroups: TargetGroupProps [],
+}
+
+/**
+ * The interface for creating and attaching a target group.
+ */
+export interface TargetGroupProps {
+  /**
+   * ID for a target group to be created.
+   */
+  readonly targetGroupID: string,
+  /**
+   * Properties for adding new targets to a listener.
+   */
+  readonly addTargetGroupProps: elbv2.AddApplicationTargetsProps | elbv2.AddNetworkTargetsProps,
+  /**
+   * Listener to attach to.
+   */
+  readonly listener: elbv2.IApplicationListener | elbv2.INetworkListener
+}
+```
+
+### Part 3: Add new API
+
+``` ts
+/**
+ * The base class for Ec2Service and FargateService services.
+ */
+export abstract class BaseService extends Resource
+  implements IService, elbv2.IApplicationLoadBalancerTarget, elbv2.INetworkLoadBalancerTarget {
+
+  ...
+
+  public attachToTargetGroup(targetGroup: elbv2.ApplicationTargetGroup | elbv2.NetworkTargetGroup,
+                             target: ContainerTarget): elbv2.LoadBalancerTargetProps {
+    if (targetGroup instanceof elbv2.ApplicationTargetGroup) {
+      const ret = this.attachToELB(targetGroup, target);
+
+      // Open up security groups. For dynamic port mapping, we won't know the port range
+      // in advance so we need to open up all ports.
+      const port = this.taskDefinition.defaultContainer!.ingressPort;
```

Comment:
This will need to be updated to reflect the desired host port for the container target.